### ISSUE TITLE
Feature request regarding .templates (overridden)?

### ISSUE THREAD

**Issue by simonlock** [gcgarner/IOTstack#128]
Hi Graham

Would it be feasible to introduce something like an "override" folder within .templates folder which would be used at a higher priority than the default templates found within the .templates folder when creating docker containers? And to include this folder of overridden templates in the automated backups?

I've made several changes to the defaults found in .templates that you provide i.e. different images, networking and ports and I find myself having to redo my configuration when IoTstack is updated. I think this would be a very useful enhancement and should cause no issues for the current users of IOTstack. Just an idea.

Best regards

Simon

**Comment by Slyke** [comment]
I think maybe a yaml merging type code would be better suited for this type of thing? That way it can inject properties under specific objects, while still providing updates where necessary.

I could probably code something like this, could use Python or NodeJS for it. Python is probably better as it comes installed already in most systems (especially Raspberry Pis).

**Comment by simonlock** [comment]
I agree. The merging sound like an excellent idea. That would allow for the overriding of components of each template file, therefore allowing customisation, whilst not having to replace the template files completely.

**Comment by Slyke** [comment]
Found some example code that will achieve this: https://stackoverflow.com/questions/47424865/merge-two-yaml-files-in-python

**Comment by simonlock** [comment]
Excellent. This sounds like the perfect solution.
